cc/crypto/hpke/recipient_context.cc

```diff
@@ -16,7 +16,180 @@
+      plaintext_bytes.size() + EVP_AEAD_max_overhead(EVP_HPKE_AEAD_aead(EVP_hpke_aes_256_gcm()));
+
+  std::vector<uint8_t> ciphertext_bytes(max_out_len);
+  size_t ciphertext_bytes_len;
+
+  if (!EVP_AEAD_CTX_seal(
+          /* ctx= */ aead_response_context_.get(),
+          /* out= */ ciphertext_bytes.data(),
+          /* out_len= */ &ciphertext_bytes_len,
+          /* max_out_len= */ ciphertext_bytes.size(),
+          /* nonce= */ response_nonce_.data(),
+          /* nonce_len= */ response_nonce_.size(),
+          /* in= */ plaintext_bytes.data(),
+          /* in_len= */ plaintext_bytes.size(),
+          /* ad= */ associated_data_bytes.data(),
+          /* ad_len= */ associated_data_bytes.size())) {
+    return absl::AbortedError("Unable to seal response messge");
+  }
+  ciphertext_bytes.resize(ciphertext_bytes_len);
+  std::string ciphertext(ciphertext_bytes.begin(), ciphertext_bytes.end());
+  return ciphertext;
+}
+
+RecipientResponseContext::~RecipientResponseContext() {
+  EVP_AEAD_CTX_free(aead_response_context_.release());
+}
+
+absl::StatusOr<RecipientContext> SetupBaseRecipient(
+    absl::string_view serialized_encapsulated_public_key, KeyPair recipient_key_pair,
+    absl::string_view info) {
+  // First verify that the supplied key pairing is valid using the BoringSSL library.
+  std::vector<uint8_t> private_key_bytes(recipient_key_pair.private_key.begin(),
+                                         recipient_key_pair.private_key.end());
+  std::vector<uint8_t> public_key_bytes(recipient_key_pair.public_key.begin(),
+                                        recipient_key_pair.public_key.end());
+  std::vector<uint8_t> encap_public_key_bytes(serialized_encapsulated_public_key.begin(),
+                                              serialized_encapsulated_public_key.end());
+
+  bssl::ScopedEVP_HPKE_KEY recipient_keys;
+
+  absl::Status keys_valid_status = ValidateKeys(public_key_bytes, private_key_bytes,
+                                                encap_public_key_bytes, recipient_keys.get());
+  if (!keys_valid_status.ok()) {
+    return keys_valid_status;
+  }
+  std::vector<uint8_t> info_bytes(info.begin(), info.end());
+
+  std::unique_ptr<EVP_HPKE_CTX> hpke_recipient_context(EVP_HPKE_CTX_new());
+  if (hpke_recipient_context == nullptr) {
+    return absl::AbortedError("Unable to generate HPKE sender context");
+  }
+
+  if (!EVP_HPKE_CTX_setup_recipient(
+          /* ctx= */ hpke_recipient_context.get(),
+          /* key= */ recipient_keys.get(),
+          /* kdf= */ EVP_hpke_hkdf_sha256(),
+          /* aead= */ EVP_hpke_aes_256_gcm(),
+          /* enc= */ encap_public_key_bytes.data(),
+          /* enc_len= */ encap_public_key_bytes.size(),
+          /* info= */ info_bytes.data(),
+          /* info_len= */ info_bytes.size())) {
+    return absl::AbortedError("Unable to setup recipient context.");
+  }
+
+  // Set up response encryption.
+  std::cout << "GETTING RESPONSE CONTEXT" << std::endl;
```

Comment:
Could you please remove logging from the library?

Reply:
Good catch, accidentally left this in